runtime/util/resolvehelp.c

```diff
@@ -58,17 +59,17 @@ getMethodForSpecialSend(J9VMThread *vmStruct, J9Class *currentClass, J9Class *re
 
 			if (vTableIndex != 0) {
 				J9Class *superclass = currentClass->superclasses[currentDepth - 1];
-				/* CMVC 170457: Algorithm for invokespecial lookup is wrong.
-				 * New Algorithm:
-				 * 1) Find the vtable index for J9Method in the resolved class.
-				 * 2) Using the vtable index to get the J9Method at that index in the current class.
-				 * 3) Walk the current class vtable backwards to find the most "recent" override of that method
-				 * 4) Lookup the method at vtableIndex in currentClass's super class
-				 */
-				method = *(J9Method **)(((UDATA)currentClass) + vTableIndex);
-				vTableIndex = vmFuncs->getVTableIndexForMethod(method, currentClass, vmStruct);
 
 				if (isInterfaceMethod) {
+					/* CMVC 170457: Algorithm for invokespecial lookup is wrong.
+					 * New Algorithm:
+					 * 1) Find the vtable index for J9Method in the resolved class.
+					 * 2) Using the vtable index to get the J9Method at that index in the current class.
+					 * 3) Walk the current class vtable backwards to find the most "recent" override of that method
+					 * 4) Lookup the method at vtableIndex in currentClass's super class
+					 */
+					method = *(J9Method **)(((UDATA)currentClass) + vTableIndex);
+					vTableIndex = vmFuncs->getVTableIndexForMethod(method, currentClass, vmStruct);
```

Comment:
The definition of `superVTableSize` below needs to be hoisted up as this isn't valid c89.